core/embed/rust/src/util.rs

```diff
@@ -94,3 +96,12 @@ impl<T, E> ResultExt for Result<T, E> {
         }
     }
 }
+
+/// Concatenates arbitrary amount of slices into a String
+pub fn format_string<const M: usize, const N: usize>(slices: [&str; M]) -> String<N> {
```

Comment:
this would be much better as a macro
```
macro_rules! build_string {
    ($max:expr, $($string:expr),+) => {
        {
            let new_string = String::<$max>::new();
            $(new_string.push_str($string).unwrap();)+
            new_string
        }
    }
}

let string = build_string!(100, "hello", "world", number.as_str());
```

Reply:
Done in [4630051](https://github.com/trezor/trezor-firmware/pull/2277/commits/4630051f31d4822486d01ac0fb02eb31dbd62845)